client/sftp_client.py

```diff
@@ -167,23 +537,71 @@ def _get_popup_menu(self):
 
 		self.popup_menu.show_all()
 
-	def _signal_treeview_button_pressed(self, _, event):
+	def _rename_selection(self):
+		selection = self.treeview.get_selection()
+		_, treeiter = selection.get_selected()
+		self.rename(treeiter)
+
+	def get_is_folder(self, fullname):
+		return stat.S_ISDIR(self.stat(fullname).st_mode)
+
+	def get_file_size(self, fullname, stat_override=None):
+		if stat_override is not None:
+			return stat_override(fullname).st_size
+		return self.stat(fullname).st_size
+
+	def signal_tv_button_press(self, _, event):
 		if event.button == Gdk.BUTTON_SECONDARY:
 			self.popup_menu.popup(None, None, None, None, event.button, Gtk.get_current_event_time())
 			return True
 		return
 
-	def signal_expand_row(self, _, treeiter, treepath):
-		new_path = self._tvmodel[treeiter][2]  # pylint: disable=unsubscriptable-object
-		self.load_dirs(new_path, treeiter)
-		self._tvmodel.remove(self._tvmodel.iter_children(treeiter))
+	def signal_tv_key_press(self, treeview, event):
+		if event.type != Gdk.EventType.KEY_PRESS:
+			return
+		keyval = event.get_keyval()[1]
+		if keyval == Gdk.KEY_F2:
+			self._rename_selection()
+		elif keyval == Gdk.KEY_F5:
+			self.refresh()
+		elif keyval == Gdk.KEY_Delete:
+			self._delete_selection()
 
-	def signal_collapse_row(self, _, treeiter, treepath):
-		current = self._tvmodel.iter_children(treeiter)
+	def rename(self, treeiter):
+		path = self._tv_model.get_path(treeiter)
+		fullname = self._tv_model[path][2]
+		parent = self._tv_model.iter_parent(treeiter)
+		if parent is None:
+			parent_name = self.default_directory
+			self.parentless = True
+		else:
+			parent_name = self._tv_model[parent][2]
+			self.parentless = False
+		if self._tv_model[treeiter][2] is not None:
+			perm = self._check_perm(parent_name)
+			w_perm = True if perm[4] == 'w' else False
+			if not w_perm:
```

Comment:
This should just be `if perm[4] != 'w':`, making the line before it unnecessary.